use-cases/virgo/trainer.py

```diff
@@ -108,26 +110,86 @@ def create_model_loss_optimizer(self) -> None:
             self.model, self.optimizer, **distribute_kwargs
         )
 
+    def create_dataloaders(self, train_dataset: Dataset, validation_dataset: Dataset | None = None, test_dataset: Dataset | None = None) -> None:
+        """
+        Override the create_dataloaders function to use the custom_collate function. 
+        """
+        self.train_dataloader = self.strategy.create_dataloader(
+            dataset=train_dataset,
+            batch_size=self.config.batch_size,
+            num_workers=self.config.num_workers_dataloader,
+            pin_memory=self.config.pin_gpu_memory,
+            generator=self.torch_rng,
+            shuffle=self.config.shuffle_train,
+            collate_fn=self.custom_collate
+        )
+        if validation_dataset is not None:
+            self.validation_dataloader = self.strategy.create_dataloader(
+                dataset=validation_dataset,
+                batch_size=self.config.batch_size,
+                num_workers=self.config.num_workers_dataloader,
+                pin_memory=self.config.pin_gpu_memory,
+                generator=self.torch_rng,
+                shuffle=self.config.shuffle_validation,
+                collate_fn=self.custom_collate
+            )
+        if test_dataset is not None:
+            self.test_dataloader = self.strategy.create_dataloader(
+                dataset=test_dataset,
+                batch_size=self.config.batch_size,
+                num_workers=self.config.num_workers_dataloader,
+                pin_memory=self.config.pin_gpu_memory,
+                generator=self.torch_rng,
+                shuffle=self.config.shuffle_test,
+                collate_fn=self.custom_collate
+            )
+
+    def custom_collate(self, batch):
+        """
+        Custom collate function to concatenate input tensors along their first dimension.
+        """
+        # Some batches contain None values, if any files from the dataset did not match the criteria
+        # (i.e. three auxilliary channels)
+        batch = list(filter(lambda x: x is not None, batch))
```

Comment:
While "filter()" is fine functionality-wise, it is generally not considered very "Pythonic". Usually the more Pythonic way of doing this would be with a list generator, i.e. [a for a in ... if a ...]